test/test_mypy.py

```diff
@@ -14,22 +14,21 @@
 
 """Test that each file in mypy_fails/ actually fails mypy, and test some
 sample client code that uses PyMongo typings."""
-
 import os
 import tempfile
 import unittest
 from typing import TYPE_CHECKING, Any, Dict, Iterable, Iterator, List
 
 try:
-    from typing import TypedDict  # type: ignore[attr-defined]
-
     # Not available in Python 3.7
+    from typing_extensions import TypedDict
+
```

Comment:
Let's revert these changes and go back to using typing.

Reply:
We will need to keep using `typing_extensions` in this file, or else split off a different file to run `mypy` on Python 3.7.  The reason it was working before is that we were inadvertently ignoring errors when running this file.

Reply:
> The reason it was working before is that we were inadvertently ignoring errors when running this file.

Can you explain? How was this fixed?

Reply:
We added a new invocation in GitHub Actions in this PR to run this file without disabling several error checks.

Reply:
I see thanks.